mbedtls-sys/build/bindgen.rs

```diff
@@ -8,78 +8,136 @@
-        File::create(&header)
-            .and_then(|mut f| {
-                Ok(for h in headers::enabled_ordered() {
-                    writeln!(f, "#include <mbedtls/{}>", h)?;
-                })
-            }).expect("bindgen-input.h I/O error");
+        let mut input = String::new();
+        for h in headers::enabled_ordered() {
+            let _ = writeln!(input, "#include <mbedtls/{}>", h);
+        }
+
+        let mut cc = cc::Build::new();
+        cc.include(&self.mbedtls_include)
+        .flag(&format!(
+            "-DMBEDTLS_CONFIG_FILE=\"{}\"",
+            self.config_h.to_str().expect("config.h UTF-8 error")
+        ));
 
-        let include = self.mbedtls_src.join("include");
+        for cflag in &self.cflags {
+            cc.flag(cflag);
+        }
+
+        let mut bindgen = bindgen::builder();
+        for arg in cc.get_compiler().args() {
+            bindgen = bindgen.clang_arg(arg.to_str().unwrap());
+        }
 
-        let logger = StderrLogger;
-        let mut bindgen = bindgen::Builder::new(header.into_os_string().into_string().unwrap());
         let bindings = bindgen
-            .log(&logger)
-            .clang_arg("-Dmbedtls_t_udbl=mbedtls_t_udbl;") // bindgen can't handle unused uint128
-            .clang_arg(format!(
-                "-DMBEDTLS_CONFIG_FILE=\"{}\"",
-                self.config_h.to_str().expect("config.h UTF-8 error")
-            )).clang_arg(format!(
-                "-I{}",
-                include.to_str().expect("include/ UTF-8 error")
-            )).match_pat(include.to_str().expect("include/ UTF-8 error"))
-            .match_pat(self.config_h.to_str().expect("config.h UTF-8 error"))
-            .use_core(true)
-            .derive_debug(false) // buggy :(
-            .ctypes_prefix(vec!["types".to_owned(), "raw_types".to_owned()])
-            .remove_prefix("mbedtls_")
-            .rust_enums(false)
-            .convert_macros(true)
-            .macro_int_types(
-                vec![
-                    "sint",
-                    "sint",
-                    "sint",
-                    "slonglong",
-                    "sint",
-                    "sint",
-                    "sint",
-                    "slonglong",
-                ].into_iter(),
-            ).generate()
-            .expect("bindgen error");
+            .header_contents("bindgen-input.h", &input)
+            .allowlist_function("^(?i)mbedtls_.*")
+            .allowlist_type("^(?i)mbedtls_.*")
+            .allowlist_var("^(?i)mbedtls_.*")
+            .allowlist_recursively(false)
+            .blocklist_type("^mbedtls_time_t$")
+            .use_core()
+            .ctypes_prefix("::types::raw_types")
+            .parse_callbacks(Box::new(MbedtlsParseCallbacks))
+            .default_enum_style(bindgen::EnumVariation::Consts)
+            .generate_comments(false)
+            .derive_copy(true)
+            .derive_debug(false) // used to be buggy, may be possible now?
```

Comment:
Seems to work for `x86_64-fortanix-unknown-sgx` but for `x86_64-unknown-linux-gnu` it results in:
`libc::pthread_mutex_t` cannot be formatted using `{:?}` because it doesn't implement `Debug`. I would remove the comment.